### DISCUSSION TITLE
How to pass array of JS Objects for a 1-n relation

### DISCUSSION THREAD

**Discussion by ghost** [prisma/prisma#14803] [Q&A]
Using the GitHub API to build a list of repositories in cloud environment and the teams which have access to that repository. The model looks like:

```
model Repo {
  name                                 String   @id @map("repo_name")
  teams Team[]

  @@map("repo")
}

model Team {
  name String @id
  repo Repo? @relation(fields: [repoName], references: [name])
  repoName String @map("repo_name")

@@map("team")
}
```

I want to build the team array ahead of time and just pass that to the create function and am getting absolutely. nowhere. Actually, more specifically I'm using upsert and getting nowhere in the create call there.  Here is the snipping of the call (I actually have more fields here but kept them out to keep the example simple):

```
        const newRepo = await prisma.repo.upsert({
          where: {
            name: tempRepo.name,
          },
          update: {
            defaultBranchName: tempRepo.defaultBranchName,
          },
          create: {
            name: tempRepo.name,
            defaultBranchName: tempRepo.defaultBranchName,
            teams: { create: { name: "blah" } },
```

Here I was just hardcoding a value and this works.  But I want to build the array ahead of time as these are all API calls and I don't know how many teams have access to a repo. Someone mentioned in one post here about using Array.map to. build similar array and I tried that with this for the teams: line `teams: tempTeams.map((name) => ({ create: { team: name } })),`. Then I get an error:

```
Uncaught PrismaClientValidationError Error: 
Invalid `prisma.repo.upsert()` invocation:

{
  where: {
    name: 'test_repo
  },
  update: {
    defaultBranchName: 'main',
    numberOfReviewsRequired: 0,
    dismissStaleReviews: false,
    requireBranchesUpToDateBeforeMerging: false,
    includeAdministrators: false,
    protectionRules: true,
    exceptionList: true,
    emptyRepo: false
  },
  create: {
    name: 'test-repo',
    teams: [
      {
        create: {
          team: {
            name: 'Core Integration'
          }
        }
      },
      {
        create: {
          team: {
            name: 'Security Champions'
          }
        }
      }
    ],
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    defaultBranchName: 'main',
    numberOfReviewsRequired: 0,
    dismissStaleReviews: false,
    requireBranchesUpToDateBeforeMerging: false,
    includeAdministrators: false,
    protectionRules: true,
    exceptionList: true,
    emptyRepo: false
  }
}

Argument teams: Got invalid value 
[
  {
    create: {
      team: {
        name: 'Core Integration'
      }
    }
  },
  {
    create: {
      team: {
        name: 'Security Champions'
      }
    }
  }
]
on prisma.upsertOneRepo. Provided List<Json>, expected TeamCreateNestedManyWithoutRepoInput:
type TeamCreateNestedManyWithoutRepoInput {
  create?: TeamCreateWithoutRepoInput | List<TeamCreateWithoutRepoInput> | TeamUncheckedCreateWithoutRepoInput | List<TeamUncheckedCreateWithoutRepoInput>
  connectOrCreate?: TeamCreateOrConnectWithoutRepoInput | List<TeamCreateOrConnectWithoutRepoInput>
  createMany?: TeamCreateManyRepoInputEnvelope
  connect?: TeamWhereUniqueInput | List<TeamWhereUniqueInput>
}
    at validate (undefined:27354:20)
    at serializationFn (undefined:29815:19)
    at runInChildSpan (undefined:23564:12)
    at _executeRequest (undefined:29822:31)

```
Totally clueless as to how to get this to work.

**Comment by austincrim** [comment]
Hey @jasonekratz 👋,

I think you just have a bit of nesting off, this snippet works for me:
```ts
let result = await prisma.repo.upsert({
  where: {
    name: 'test'
  },
  update: {
    emptyRepo: false
  },
  create: {
    name: 'test',
    emptyRepo: true,
    teams: {
      create: [
        {
          name: 'team 1'
        },
        {
          name: 'team 2'
        }
      ]
    }
  }
})
```

Under `create.teams`, you need a top-level `create` property, then you can pass the array of teams to that property.

Let me know if that doesn't work!

**Reply by ghost** [reply]
Thank you @austincrim !  Will give that a shot and let you know. Been pulling my hair out over this heh. Long time developer but brand new at the JS/TS thing.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
